tests/framework/db/QueryTest.php

```diff
@@ -110,6 +110,60 @@ public function testFilterWhere()
         $this->assertEquals($condition, $query->where);
     }
 
+    public function testFilterHaving()
+    {
+        // should work with hash format
+        $query = new Query;
+        $query->filterHaving([
+            'id' => 0,
+            'title' => '   ',
+            'author_ids' => [],
+        ]);
+        $this->assertEquals(['id' => 0], $query->having);
+
+        $query->andFilterHaving(['status' => null]);
+        $this->assertEquals(['id' => 0], $query->having);
+
+        $query->orFilterHaving(['name' => '']);
+        $this->assertEquals(['id' => 0], $query->having);
+
+        // should work with operator format
```

Comment:
this seems the start of a new test scenario? How should one interpretate this in the future?